### ISSUE TITLE
Using RecursivePartial from common.ts in application

### ISSUE THREAD

**Issue by markus-ebner** [dash14/v-network-graph#10]
Hi, 

when using your awesome library with TS, I run into a warning in my IDE because of missing type declarations. 
I'm using the RecursivePartial to properly type the main configuration. And since it is not referenced in the index.d.ts TS does not know the type declarations. 

Is it possible to add common/common.ts to the exported type information? 

```
// @ts-ignore because of Cannot find module 'v-network-graph/lib/types/common/common' or its corresponding type declarations.
import { RecursivePartial } from 'v-network-graph/lib/types/common/common'
...
const mainConfig: UnwrapNestedRefs<RecursivePartial<Configs>> = reactive({})
```

**Comment by dash14** [comment]
Hi @markus-ebner,

Thank you for your report.
First, is it possible to use `UserConfigs` as an alias for `RecursivePartial<Configs>` ? I think `UserConfigs` has already been exported.

```ts
import { UserConfigs } from 'v-network-graph'
...
const mainConfig: UnwrapNestedRefs<UserConfigs> = reactive({})
```

If you have a case where you want to use `RecursivePartial` for something other than `Configs`, please let me know again so I will modify it to export.

Best regards,

**Comment by markus-ebner** [comment]
Thanks for the fast response @dash14!
Yes, i tried it but i get the following problem with `UserConfigs`:
```
const mainConfig: UnwrapNestedRefs<UserConfigs> = reactive({
    view: {
      // TS2322: Type 'false' is not assignable to type 'true'.
      panEnabled: false,
      // TS2322: Type 'false' is not assignable to type 'true'.
      zoomEnabled: false,
      ...
    }
}
```
Shouldn't this be a boolean type?

**Comment by dash14** [comment]
Hi @markus-ebner,
Thank you very much for your reply and report!
This was a bug...
The type definition was such that false could not be specified for `view.{panEnabled, zoomEnabled}`.

`edge.summarized.*` config did not support specifying a function, but this has been implemented along with the bug fix.
Note that `edge.summarized.*` are the configs when multiple edges are merged into one, so the specified function's argument is not 'edge' but 'edges' (`{ edgeId: edge }` object).

```ts
const mainConfig = reactive<UserConfigs>({
   ...
   edge: {
      summarized: {
          shape: {
            strokeDasharray: edges => Object.values(edges)[0].someParameterOfEdge as number ?? 0
           ...
         }
     ... 
     }
  }
}
```

I just released a new version ([v0.3.2](https://github.com/dash14/v-network-graph/releases/tag/v0.3.2)), please check it.

**Comment by markus-ebner** [comment]
Thank you very much. It works as expected.
